builtin/logical/transit/path_byok_test.go

```diff
@@ -0,0 +1,225 @@
+	}
+
+	// Import the wrapping key.
+	wrapKeyImportReq := &logical.Request{
+		Path:      "keys/wrapper/import",
+		Operation: logical.UpdateOperation,
+		Storage:   s,
+		Data: map[string]interface{}{
+			"public_key": resp.Data["public_key"],
+			"type":       "rsa-4096",
+		},
+	}
+	resp, err = b.HandleRequest(context.Background(), wrapKeyImportReq)
+	if err != nil || (resp != nil && resp.IsError()) {
+		t.Fatalf("resp: %#v\nerr: %v", resp, err)
+	}
+
+	// Export the key
+	backupReq := &logical.Request{
+		Path:      "byok-export/wrapper/test-source",
+		Operation: logical.ReadOperation,
+		Storage:   s,
+	}
+	resp, err = b.HandleRequest(context.Background(), backupReq)
+	if err != nil || (resp != nil && resp.IsError()) {
+		t.Fatalf("resp: %#v\nerr: %v", resp, err)
+	}
+	keys := resp.Data["keys"].(map[string]string)
+
+	// Import the key to a new name.
+	restoreReq := &logical.Request{
+		Path:      "keys/test/import",
+		Operation: logical.UpdateOperation,
+		Storage:   s,
+		Data: map[string]interface{}{
+			"ciphertext": keys["1"],
+			"type":       keyType,
+		},
+	}
+	resp, err = b.HandleRequest(context.Background(), restoreReq)
+	if err != nil || (resp != nil && resp.IsError()) {
+		t.Fatalf("resp: %#v\nerr: %v", resp, err)
+	}
+
+	plaintextB64 := "dGhlIHF1aWNrIGJyb3duIGZveA==" // "the quick brown fox"
+	// Perform encryption, signing or hmac-ing based on the set 'feature'
+	var encryptReq, signReq, hmacReq *logical.Request
+	var ciphertext, signature, hmac string
+	switch feature {
+	case "encrypt-decrypt":
+		encryptReq = &logical.Request{
+			Path:      "encrypt/test-source",
+			Operation: logical.UpdateOperation,
+			Storage:   s,
+			Data: map[string]interface{}{
+				"plaintext": plaintextB64,
+			},
+		}
+		resp, err = b.HandleRequest(context.Background(), encryptReq)
+		if err != nil || (resp != nil && resp.IsError()) {
+			t.Fatalf("resp: %#v\nerr: %v", resp, err)
+		}
+		ciphertext = resp.Data["ciphertext"].(string)
+
+	case "sign-verify":
+		signReq = &logical.Request{
+			Path:      "sign/test-source",
+			Operation: logical.UpdateOperation,
+			Storage:   s,
+			Data: map[string]interface{}{
+				"input": plaintextB64,
+			},
+		}
+		resp, err = b.HandleRequest(context.Background(), signReq)
+		if err != nil || (resp != nil && resp.IsError()) {
+			t.Fatalf("resp: %#v\nerr: %v", resp, err)
+		}
+		signature = resp.Data["signature"].(string)
+
+	case "hmac-verify":
+		hmacReq = &logical.Request{
+			Path:      "hmac/test-source",
+			Operation: logical.UpdateOperation,
+			Storage:   s,
+			Data: map[string]interface{}{
+				"input": plaintextB64,
+			},
+		}
+		resp, err = b.HandleRequest(context.Background(), hmacReq)
+		if err != nil || (resp != nil && resp.IsError()) {
+			t.Fatalf("resp: %#v\nerr: %v", resp, err)
+		}
+		hmac = resp.Data["hmac"].(string)
+	}
+
+	// validationFunc verifies the ciphertext, signature or hmac based on the
+	// set 'feature'
+	validationFunc := func(keyName string) {
```

Comment:
is there a reason this isn't a function external to this call?

Reply:
I think it uses variables from above and this follows the existing pattern of tests above.

Reply:
Er, and by above, I mean `path_backup_test.go` lol :-D